lib/model/rwfolder.go

```diff
@@ -87,6 +89,8 @@ type rwFolder struct {
 	copiers      int
 	pullers      int
 	shortID      uint64
+	newVersion   protocol.Vector
```

Comment:
Yes, but `p` is shared between a bunch of goroutines, while your `newVersion` and `curVersion` should be per file.